src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusMetric.cs

```diff
@@ -41,10 +42,31 @@ public PrometheusMetric(string name, string unit, PrometheusType type, bool disa
             // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L242-L246
             if (!sanitizedName.Contains(sanitizedUnit))
             {
-                sanitizedName = sanitizedName + "_" + sanitizedUnit;
+                sanitizedName += $"_{sanitizedUnit}";
             }
+
+            // OpenMetrics name MUST be suffixed with '_{unit}', regardless of whether the unit name appears within the text.
+            // Note that this may change in the future, however for the moment Prometheus will fail to read the metric using
+            // OpenMetrics format unless the suffix matches the unit.
+            // https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#unit
+            if (!openMetricsName.EndsWith(sanitizedUnit))
+            {
+                openMetricsName += $"_{sanitizedUnit}";
+            }
+        }
+
+        // Special case: Converting "1" to "ratio".
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/b2f923fb1650dde1f061507908b834035506a796/specification/compatibility/prometheus_and_openmetrics.md#L239
+        if (type == PrometheusType.Gauge && unit == "1" && !sanitizedName.Contains("ratio"))
```

Comment:
I would consider holding off on implementing this right now: https://github.com/open-telemetry/opentelemetry-specification/issues/4058.  But it is also fine to do that as a follow-up later if you want to wait for the discussion to resolve first.